### PR TITLE
feat: add Cognito auth and expand User endpoints

### LINKED ISSUE #44: [FEATURE] User Entity Basic Functionalities
- [x] Create
- [x] Update attributes (only by self)
- [x] Delete (only by self)
- [x] Get all users (only by admin)
- [x] Get a single user
- [x] (Add sample data)

### LINKED ISSUE #50: [FEATURE] Replace authentication to use Amazon Cognito

### PR DESCRIPTION
**Pull request by tg2648**
#### Authentication
- [SecurityConfig.java](https://github.com/astrohsy/ZETCH/blob/6838650b68fed758923281e0f423cc0c5be33597/api_server/src/main/java/io/zetch/app/security/SecurityConfig.java): enabling/disabling authentication for routes
- [SecurityService.java](https://github.com/astrohsy/ZETCH/blob/6838650b68fed758923281e0f423cc0c5be33597/api_server/src/main/java/io/zetch/app/security/SecurityService.java): security business logic to facilitate method-level security using `@PreAuthorize`
- See [README](https://github.com/astrohsy/ZETCH/tree/auth/cognito-2/api_server#auth) for details on getting auth tokens in Postman or Swagger UI

#### User endpoints
- Getting all users requires the caller to be an admin
- Users cannot edit and delete other users 

#### Sample data
- When the application starts, some sample User data is added to the database. All of them are already in Cognito, so you can login as them too:
https://github.com/astrohsy/ZETCH/blob/6838650b68fed758923281e0f423cc0c5be33597/api_server/src/main/java/io/zetch/app/ZetchApplication.java#L38-L47

Closes #50
Closes #44

### CODE REVIEW

**Comment by tg2648** [comment]
@zagbapu Here is where I added sample users. Let me know if there is a better/cleaner way! 
https://github.com/astrohsy/ZETCH/blob/6838650b68fed758923281e0f423cc0c5be33597/api_server/src/main/java/io/zetch/app/ZetchApplication.java#L38-L47

File: api_server/src/test/java/io/zetch/app/domain/AffiliationTest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class AffiliationTest {
+
+  @ParameterizedTest
+  @CsvSource({"student, STUDENT", "faculty, FACULTY", "other, OTHER", "admin, ADMIN"})
```
Comment (astrohsy) [inline api_server/src/test/java/io/zetch/app/domain/AffiliationTest.java:18]: This one is so fascinating!

File: api_server/src/main/java/io/zetch/app/web/HelloController.java [hunk 2]
```diff
@@ -1,13 +1,27 @@
 package io.zetch.app.web;
 
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 public class HelloController {
```
Comment (astrohsy) [inline api_server/src/main/java/io/zetch/app/web/HelloController.java:8]: Is this controller for just testing??

**Review by astrohsy — APPROVED** [review]
💯  Thanks for the great work!

File: api_server/src/main/java/io/zetch/app/web/HelloController.java (hunk 2 above)
Comment (tg2648) [reply api_server/src/main/java/io/zetch/app/web/HelloController.java:8]: Yes, I kept it in as an example of public/private routes. We can clean up later

**Pull request merged**